src/SFML/Window/DRM/DRMContext.cpp

```diff
@@ -394,56 +381,56 @@ namespace
 
         if (encoder)
         {
-            drm.crtcId = encoder->crtc_id;
+            drmNode.crtcId = encoder->crtc_id;
         }
         else
         {
-            const sf::Uint32 crtcId = findCrtcForConnector(drm, *resources, *connector);
+            const sf::Uint32 crtcId = findCrtcForConnector(drmNode, *resources, *connector);
             if (crtcId == 0)
             {
                 sf::err() << "No crtc found!" << std::endl;
                 return -1;
             }
 
-            drm.crtcId = crtcId;
+            drmNode.crtcId = crtcId;
         }
 
         drmModeFreeResources(resources);
 
-        drm.connectorId = connector->connector_id;
+        drmNode.connectorId = connector->connector_id;
 
-        drm.savedConnector = connector;
-        drm.savedEncoder = encoder;
+        drmNode.savedConnector = connector;
+        drmNode.savedEncoder = encoder;
 
         // Get original display mode so we can restore display mode after program exits
-        drm.originalCrtc = drmModeGetCrtc(drm.fileDescriptor, drm.crtcId);
+        drmNode.originalCrtc = drmModeGetCrtc(drmNode.fileDescriptor, drmNode.crtcId);
 
-        // Let's use the current mode rather than the preferred one if the user didn't specify a mode with env vars
-        if (!drm.mode)
-        {
-#ifdef SFML_DEBUG
-            sf::err() << "DRM using the current mode" << std::endl;
-#endif
-            drm.mode = &(drm.originalCrtc->mode);
-        }
+        gbmDevice = gbm_create_device(drmNode.fileDescriptor);
 
-#ifdef SFML_DEBUG
-        sf::err() << "DRM Mode used: " << drm.mode->name << "@" << drm.mode->vrefresh << std::endl;
-#endif
+        std::atexit(cleanup);
+        initialized = true;
 
-        return 0;
+        pollFD.fd = drmNode.fileDescriptor;
+        pollFD.events = POLLIN;
+        drmEventCtx.version = 2;
+        drmEventCtx.page_flip_handler = pageFlipHandler;
+
+        drmNode.mode = 0;
+
+        return 1;
     }
 
-    void checkInit()
+    void setDrmMode( unsigned int width=0, unsigned int height=0 )
     {
-        if (initialized)
+	// don't do anything if supplied width and height are 0 and we already have a drm mode
+	if (( width == 0 ) && drmNode.mode )
```

Comment:
```suggestion
        if (( width == 0 ) && drmNode.mode )
```
Please use spaces to indent instead of tabs.